services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/client_api.py

```diff
@@ -258,17 +262,39 @@ async def get_entrypoint_container_name(
             f"/{self.API_VERSION}/containers/name?filters={filters}",
         )
         try:
-            async with httpx.AsyncClient(timeout=self._base_timeout) as client:
-                response = await client.get(url=url)
-                if response.status_code == status.HTTP_404_NOT_FOUND:
-                    raise EntrypointContainerNotFoundError()
-                response.raise_for_status()
+            response = await self._client.get(url=url)
+            if response.status_code == status.HTTP_404_NOT_FOUND:
+                raise EntrypointContainerNotFoundError()
+            response.raise_for_status()
 
-                return response.json()
+            return response.json()
         except httpx.HTTPError:
             log_httpx_http_error(url, "GET", traceback.format_exc())
             raise
 
+    async def restart_containers(self, dynamic_sidecar_endpoint: str) -> None:
+        """
+        runs docker-compose stop and docker-compose start in succession
+        resulting in a container restart without loosing state
+        """
+        url = get_url(
+            dynamic_sidecar_endpoint, f"/{self.API_VERSION}/containers:restart"
+        )
+        try:
+            response = await self._client.post(
```

Comment:
I find using a single client a good improvement 🎉 
I still think the error handling has quite some room for improvement. we should leave this a  design session